fe/fe-core/src/main/java/com/starrocks/sql/optimizer/CTEUtils.java

```diff
@@ -25,70 +24,50 @@
 import com.starrocks.sql.optimizer.statistics.Statistics;
 import com.starrocks.sql.optimizer.statistics.StatisticsCalculator;
 
-import java.util.LinkedList;
+import java.util.BitSet;
 
 public class CTEUtils {
-
     public static void collectCteOperators(OptExpression anchor, OptimizerContext context) {
         context.getCteContext().reset();
-        collectCteProduce(anchor, context);
-        collectCteConsume(anchor, context);
+        collectCteOperatorsImpl(anchor, context, new BitSet());
     }
 
-    /*
-     * Estimate the complexity of the produce plan
-     * */
-    private static void collectCteProduce(OptExpression root, OptimizerContext context) {
-        for (OptExpression child : root.getInputs()) {
-            collectCteProduce(child, context);
-        }
-
-        if (OperatorType.LOGICAL_CTE_PRODUCE.equals(root.getOp().getOpType())) {
+    public static void collectCteOperatorsImpl(OptExpression root, OptimizerContext context, BitSet searchConsume) {
```

Comment:
I haven't see where `searchConsume` is updated

Reply:
line 45、48